pkg/odo/genericclioptions/runnable.go

```diff
@@ -93,6 +93,11 @@ func GenericRun(o Runnable, cmd *cobra.Command, args []string) {
 		klog.V(4).Infof("WARNING: debug telemetry, if enabled, will be logged in %s", debugTelemetry)
 	}
 
+	err = scontext.SetCaller(cmd.Context(), scontext.CallerType(os.Getenv(segment.TelemetryCaller)))
+	if err != nil {
+		log.Warningf("error handling caller property for telemetry: %v", err)
```

Comment:
@kadel Not sure what the behavior should be for example if an invalid / unknown value is set as `caller`. I chose to just display a warning (and not block the `odo` command from running), but telemetry will not contain the `caller` property. Is that ok?

Reply:
My concern about this approach is that it will create bad data in our telemetry. 

In case of the wrong `TELEMETRY_CALLER` value, I would either error out (block odo from running) or show a warning, but in that case, I think we should set `caller` property.

The reason for this is that we can filter out wrong `caller` values in telemetry, but if `caller` is omited, then it will be automatically assigned as a regular cli call, it will affect statistics, and we won't be able to filter those out.

Reply:
I think it's right to not block odo from running and simply display warning, but I would prefer if we do not log it as a level 0, user-facing warning, given that we do not want users to know of telemetry. IMO, if the value is not acceptable, we can return "caller: unknown".

Reply:
I also agree with not blocking `odo`.
But maybe we want to know which value was set, instead of overriding it with `unknown`?
Given the explicit list of allowed values, we could still be able to filter.

Reply:
> but I would prefer if we do not log it as a level 0, user-facing warning, given that we do not want users to know of telemetry.

If `TELEMETRY_CALLER` is explicitly set, it may mean to me that the user is "telemetry-aware".. In this case, I think it might make sense to tell them that the value is not acceptable, no?

Reply:
Okay, so I've changed the behavior so that the `caller` property is set in telemetry with the value of `TELEMETRY_CALLER`, no matter what this value is. We are still displaying a user-facing warning if `TELEMETRY_CALLER` is invalid, but the `odo` command does not error out.

See [5d731b67810eea8b82baf518906768f013340f8c](https://github.com/redhat-developer/odo/pull/6223/commits/5d731b67810eea8b82baf518906768f013340f8c)

I've updated the integration tests accordingly in [515b8a3e4a01f9fe565cc5217d79e3c823c90890](https://github.com/redhat-developer/odo/pull/6223/commits/515b8a3e4a01f9fe565cc5217d79e3c823c90890)